src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -761,6 +761,57 @@ private IEnumerable<Element> GetElementSubComponents(Autodesk.Revit.DB.Element e
             return components;
         }
 
+        /// <summary>
+        /// Gets the parent component of the Element.
+        /// </summary>
+        /// <returns>Parent Element</returns>
+        public Element GetParentComponent()
+        {
+            var parentComponent = GetParentElementFromElement(this.InternalElement);
+            return parentComponent;
+        }
+
+        private Element GetParentElementFromElement(Autodesk.Revit.DB.Element element)
+        {
+            Autodesk.Revit.DB.Element parent;
+            BuiltInCategory builtInCategory = (BuiltInCategory)System.Enum.Parse(typeof(BuiltInCategory),
+                                                                                 element.Category.Id.ToString());
+            switch (builtInCategory)
+            {
+                case BuiltInCategory.OST_StairsLandings:
+                    var stairElement = element as Autodesk.Revit.DB.Architecture.StairsLanding;
+                    if (stairElement == null)
+                        throw new InvalidOperationException(Properties.Resources.NoSuperComponent);
+                    parent = stairElement.GetStairs();
```

Comment:
let's move branches of this `switch` each to a separate method, would be cleaner and easier to follow.
the statement here would then be `parent = GetParentElementForStairsLandings()`